sdk/core/azure-core/inc/azure/core/cryptography/hash.hpp

```diff
@@ -113,18 +113,22 @@ namespace Azure { namespace Core { namespace Cryptography {
 
   public:
     /**
-     * @brief Construct a default instance of @Md5Hash.
+     * @brief Construct a default instance of #Azure::Core::Cryptography::Md5Hash.
      */
-    explicit Md5Hash();
+    Md5Hash();
 
     /**
-     * @brief Cleanup any state when destroying the instance of @Md5Hash.
+     * @brief Cleanup any state when destroying the instance of #Azure::Core::Cryptography::Md5Hash.
      */
-    ~Md5Hash();
+    ~Md5Hash() override;
 
   private:
     void* m_md5Context;
 
+    // Delete the copy constructor, along with the assignment operator.
+    Md5Hash(Md5Hash const&) = delete;
+    void operator=(Md5Hash const&) = delete;
```

Comment:
@CaseyCarter  since the destructor is user declared for this class, do we still need to delete the copy constructors?

![image](https://user-images.githubusercontent.com/24213737/107483751-5a296380-6b36-11eb-9fd9-acf0d4b6c178.png)

Isn't it deprecated to be implicitly defaulted?   Just curious about this one here.

Reply:
If we apply my suggestion to delete `Hash`'s copies, then we won't need to (explicitly) delete them in `Md5Hash` since they'll be implicitly deleted by virtue of `Md5Hash`'s base class `Hash` having inaccessible-or-deleted copies.

In the status quo, I think they should be explicitly deleted because this "deprecation" is really just WG21 acknowledging that they got the defaults wrong in C++98. The corresponding behavior is unlikely to ever removed (or result in a non-off-by-default warning) due to the enormous amount of code it would break.